src/integrations.py

```diff
@@ -32,3 +42,42 @@ def load(cls, provider: AuthProxyProvider) -> "AuthProxyIntegrationData":
             authenticated_emails=authenticated_emails,
             authenticated_email_domains=authenticated_email_domains,
         )
+
+
+class TrustedCertificatesTransferIntegration:
+    def __init__(self, charm: CharmBase):
+        self._charm = charm
+        self._container = charm._container
+        self._peers = charm._peers
+        self.cert_transfer_requires = CertificateTransferRequires(charm, relationship_name="receive-ca-cert")
+
+    def update_trusted_ca_certs(self) -> None:
+        """Receive trusted certificates from the certificate_transfer integration.
+
+        This function is needed because relation events are not emitted on upgrade,
+        and because there is no persistent storage for certs.
+        """
+        if not self._charm.model.get_relation(relation_name=self.cert_transfer_requires.relationship_name):
+            logger.warning(
+                "Missing certificate_transfer integration, run `juju config oauth2-proxy-k8s dev=true` to skip validation of certificates presented when using HTTPS providers. Don't do this in production"
+            )
+            return
+
+        logger.info(
+            "Pulling trusted ca certificates from %s relation.",
+            self.cert_transfer_requires.relationship_name,
+        )
+        certs = []
+        relation = self._charm.model.get_relation(self.cert_transfer_requires.relationship_name)
+        for unit in set(relation.units).difference([self._charm.app, self._charm.unit]):
+            # Handles the case of multi-unit CA, each unit providing a different ca cert
+            cert_path = TRUSTED_CA_TEMPLATE.substitute(rel_id=relation.id, unit=unit.name)
+            if cert := relation.data[unit].get("ca"):
+                self._container.push(cert_path, cert, make_dirs=True)
```

Comment:
We don't need to push this to the container since we are pushing the ca bundle. See [here](https://github.com/canonical/kratos-operator/blob/main/src/certificate_transfer_integration.py#L52) for an example.